cdap-proto/src/main/java/io/cdap/cdap/proto/artifact/AppRequest.java

```diff
@@ -31,13 +31,14 @@
   private final ArtifactSummary artifact;
   private final T config;
   private final T configuration;
+  private final T parentVersion;
```

Comment:
why parentVersion and version here are the generic type? Shouldn't them just be String?

Reply:
yes, parentVersion can be string. I changed version to "changeSummary" which is an object since it can be extended in the future